src/main/scala/translating/GTIRBToIR.scala

```diff
@@ -0,0 +1,568 @@
+    val v = map.values.find(_.contains(value))
+    val key = v.flatMap(v => map.find(_._2 == v).map(_._1))
+    key
+  }
+
+  //Creates addresses of blks using gtirb maps
+  def create_addresses(): mutable.Map[ByteString, Int] = {
+
+    val blockAddresses: mutable.Map[ByteString, Int] = mutable.Map.empty
+
+    for {
+      mod <- mods
+      section <- mod.sections
+      byteInterval <- section.byteIntervals
+      block <- byteInterval.blocks
+      if !block.getCode.uuid.isEmpty
+    } {
+      blockAddresses += (block.getCode.uuid -> (byteInterval.address + block.offset).toInt)
+    }
+
+    blockAddresses
+
+  }
+
+  // Creates CFG map mapping blk UUIDs to a buffer of outgoing edges 
+  def create_cfg_map(): (mutable.Map[String, ArrayBuffer[Edge]], mutable.Map[String, ArrayBuffer[Edge]]) = {
+
+    val edges = cfg.edges
+    val edgeMap: mutable.Map[String, ArrayBuffer[Edge]] = mutable.Map.empty
+    val incomingEdgeMap: mutable.Map[String, ArrayBuffer[Edge]] = mutable.Map.empty
+
+    for (edge <- edges) {
+
+      val targetUuid = Base64.getUrlEncoder.encodeToString(edge.targetUuid.toByteArray)
+      val sourceUuid = Base64.getUrlEncoder.encodeToString(edge.sourceUuid.toByteArray)
+
+      if (edgeMap.contains(sourceUuid)) {
+        edgeMap(sourceUuid) += edge
+      } else {
+        edgeMap += (sourceUuid -> ArrayBuffer(edge))
+      }
+
+      if (incomingEdgeMap.contains(targetUuid)) {
+        incomingEdgeMap(targetUuid) += edge
+      } else {
+        incomingEdgeMap += (targetUuid -> ArrayBuffer(edge))
+      }
+
+    }
+    (edgeMap, incomingEdgeMap)
+  }
+
+  // Another map of blk UUIDs to Symbols related to that blk. 
+  // Symbols are usually blk names or strings related to that blk
+  def create_symMap(): mutable.Map[ByteString, mutable.Set[Symbol]] = {
+    val symMap = mutable.Map[ByteString, mutable.Set[Symbol]]()
+
+    for (sym <- symbols) {
+      if (sym.optionalPayload.isReferentUuid) {
+        val ruuid = sym.optionalPayload.referentUuid.get
+
+        if (symMap.contains(ruuid)) {
+          symMap(ruuid) += sym
+        } else {
+          symMap += (ruuid -> mutable.Set[Symbol](sym))
+        }
+
+      }
+    }
+    symMap
+  }
+
+  // Gets name of function using a UUID
+  def create_names(uuid: ByteString): String = {
+    var name: String = uuid.toString()
+
+    if (functionNames.contains(uuid)) {
+      name = symbols.find(functionNames(uuid) == _.uuid).get.name
+
+      val entryBlocks: mutable.Set[ByteString] = functionEntries(uuid)
+
+      val result = entryBlocks
+        .flatMap(e => edgeMap.getOrElse(Base64.getUrlEncoder.encodeToString(e.toByteArray), None).iterator)
+        .flatMap(elem => mods.flatMap(_.proxies).find(_.uuid == elem.targetUuid))
+        .flatMap(proxy => symMap.getOrElse(proxy.uuid, None))
+        .map(p => p.name)
+
+      if (result.nonEmpty) {
+        return result.head //. head seems weird here but i guess it works
+      }
```

Comment:
What does this section do? How do the proxy blocks come into getting the name of a function associated with a UUID?

Reply:
For weird linker functions introduced by the compiler, gtirb usually attaches the function name to a proxyblock instead of a block. In this case, I need to get the block that the proxybloxk is connected to, to find the right name of that function. This doesn't seem to be a problem for any functions the user has written themselves.

Reply:
Ah, ok. Those PLT bridging functions don't actually have names as such. When they jump to an external function that does have a name, BAP just uses that name for the bridging function too. Since DDisasm doesn't do that, I'll make it so what we're doing here is more explicit.